### DISCUSSION TITLE
How would you make a column that indicates whether note contains a certain section?

### DISCUSSION THREAD

**Discussion by johtso** [RafaelGB/obsidian-db-folder#716] [Q&A]
I'm guessing I need to use a formula column, but it's not obvious what api I'd need to use. Supposedly Obsidian's built in queries allow selecting sections..

**Comment by RafaelGB** [comment]
Hello!

I just investigate and the Obsidian API for that is metadataCache.

The formula that you need in a javascript is:
`isSectionExist.js`
```javascript
function isSectionExist(row,sectionString){
  const headings = app.metadataCache.getFileCache(row.__note__.getFile()).headings;
  return headings.some(h => h.heading===sectionString);
}

module.exports = isSectionExist
```
Formula: `${db.js.isSectionExist(row,"my section")}`

<img width="678" alt="Screenshot 2023-01-03 at 11 34 56" src="https://user-images.githubusercontent.com/11924043/210340780-fc17ee76-8707-4b2e-ad5b-2adb2f860ea8.png">

**Reply by RafaelGB** [reply]
Added to the examples: https://github.com/RafaelGB/obsidian-db-folder/tree/master/docs/examples/rowFormulas
